doc/user/content/connect-sources/postgres-amazon-rds.md

```diff
@@ -0,0 +1,395 @@
+
+{{< tabs >}}
+
+{{< tab "Allow Materialize IPs">}}
+
+
+1. In [step 2](#step-2-create-a-publication), you created a `materialize` user in your PostgreSQL database. Use the [`CREATE SECRET`](/sql/create-secret/) command to store this user's password as a secrete in Materialize:
+
+    ```sql
+    CREATE SECRET pgpass AS '<PASSWORD>';
+    ```
+
+1. Use the [`CREATE CONNECTION`](/sql/create-connection/) command to create a connection object with access and authentication details for Materialize to use:
+
+    ```sql
+    CREATE CONNECTION pg_connection TO POSTGRES (
+      HOST '<host>',
+      PORT 5432,
+      USER 'materialize',
+      PASSWORD SECRET pgpass,
+      SSL MODE 'require',
+      DATABASE '<database>'
+      );
+    ```
+
+    Replace `<host>` with the your RDS endpoint. To find your RDS endpoint, select your database in the RDS Console, and look under **Connectivity & security**.
+
+    Replace `<database>` with the name of the database containing the tables you want to replicate to Materialize.
+
+1. Use the [`CREATE SOURCE`](/sql/create-source/) command to connect Materialize to your PostgreSQL database and start ingesting data from the publication you created in PostgreSQL earlier:
+
+    ```sql
+    CREATE SOURCE mz_source
+      IN CLUSTER ingest_rds_postgres
+      FROM POSTGRES CONNECTION pg_connection (PUBLICATION 'mz_source')
+      FOR ALL TABLES;
+    ```
+
+    To ingest data from specific schemas or tables in your publication, change `FOR ALL TABLES` to `FOR SCHEMAS (<schema1>,<schema2>)` or `FOR TABLES (<table1>, <table2>)`, respectively.
+
+{{< /tab >}}
+
+{{< tab "Use PrivateLink">}}
+
+TBD
+
+{{< /tab >}}
+
+{{< tab "Use an SSH tunnel">}}
+
+TBD
+
+{{< /tab >}}
+
+{{< /tabs >}}
+
+## Step 6. Check the ingestion status
+
+At first, Materialize takes a snapshot of the relevant tables in your publication. This snapshotting process must finish before Materialize can start ingesting change events and before you can be certain that Materialize has the same view of your data as your PostgreSQL database.
+
+In this step, you'll first verify that the source is running and then check the status of the snapshotting process.
+
+1. Back in the `psql` shell connected to Materialize, use the [`mz_sources`](/sql/system-catalog/mz_catalog/#mz_sources) table to find the ID of your source:
+
+    ```sql
+    SELECT id FROM mz_sources WHERE name = 'mz_source';
+    ```
+
+1. Use the [`mz_source_statuses`](/sql/system-catalog/mz_internal/#mz_source_statuses) table to check the overall status of your source:
+
+    ```sql
+    SELECT * FROM mz_internal.mz_source_statuses
+    WHERE id = '<source_ID>';
+    ```
+
+    <p></p>
+
+    ```nofmt
+     id  |   name    |   type   |  last_status_change_at   | status  | error | details
+    -----+-----------+----------+--------------------------+---------+-------+---------
+     u61 | mz_source | postgres | 2023-05-24 18:12:35.7+00 | running |       |
+    (1 row)
+    ```
```

Comment:
This would ideally be a single step:

```sql
SELECT s.id, 
       s.name, 
       s.type, 
       status, 
       error, 
       details, 
       last_status_change_at 
FROM mz_sources s 
JOIN mz_internal.mz_source_statuses ss ON s.id = ss.id;
```

Reply:
Subsources can now error independently of the actual source (after #18539), so it's also more useful to look at source+subsources.

Reply:
The annoyance is that there might be a gazillion sources in a user's region, but then we could maybe add an `ORDER BY s.id ASC` clause.

Reply:
We annoyingly don't have a great view of which source a subsource belongs to. Filed #19534

Reply:
Once #19537 lands, you'll be able to just use this:

```sql
SELECT *
FROM mz_internal.mz_source_statuses
WHERE id =
  ANY (
    SELECT
      -- The conversion to an array is required because we
      -- don't yet support concatenating a non-array to an
      -- array
      unnest(subsources || ARRAY[id])
    FROM
      mz_sources
    WHERE
      name = '<SOURCE NAME>'
  );
```

Reply:
This can actually be done with existing features:

```sql
WITH
  source_id AS
  (SELECT id FROM mz_objects WHERE name = '<SOURCE NAME>')
SELECT *
FROM
  mz_internal.mz_source_statuses
    JOIN
      (
          SELECT referenced_object_id
          FROM mz_internal.mz_object_dependencies
          WHERE
            object_id
            = (SELECT id FROM source_id)
          UNION SELECT id FROM source_id
        )
        AS sources
      ON mz_source_statuses.id
      = sources.referenced_object_id;
```

A little annoyingly, users will have to look past the progress subsource collection to see the values they want; think maybe we ought to change the labeling of that from `subsource` to `progress`.